lib/dicon/secure_shell.ex

```diff
@@ -75,9 +80,12 @@ defmodule Dicon.SecureShell do
   end
 
   def exec(conn, command) do
-    {:ok, channel} = :ssh_connection.session_channel(conn, @timeout)
-    :success = :ssh_connection.exec(conn, channel, command, @timeout)
-    handle_reply(conn, channel, [])
+    res =
+      with {:ok, channel} <- :ssh_connection.session_channel(conn, @timeout),
+           :success <- :ssh_connection.exec(conn, channel, command, @timeout),
```

Comment:
I think we need to handle bad case from this return in `format_if_error`, it could be `:failure | : success | {:error, reason}`. :confused: